api/client/batch_client.py

```diff
@@ -30,61 +55,81 @@ def __init__(self, api_host, access_token):
 
     @gen.coroutine
     def get_data(self, url, headers, params=None):
+        base_log_record = dict(route=url, params=params)
+
+        def log_request(start_time, retry_count, msg, status_code):
+            elapsed_time = time.time() - start_time
+            log_record = dict(base_log_record)
+            log_record['elapsed_time_in_ms'] = 1000 * elapsed_time
+            log_record['retry_count'] = retry_count
+            log_record['status_code'] = status_code
+            if status_code == 200:
+                self._logger.debug(msg, extra=log_record)
+            else:
+                self._logger.warning(msg, extra=log_record)
+
         """General 'make api request' function.
 
         Assigns headers and builds in retries and logging.
         """
-        base_log_record = dict(route=url, params=params)
         retry_count = 0
         self._logger.debug(url)
 
         # append version info
         headers.update(lib.get_version_info())
 
         while retry_count < cfg.MAX_RETRIES:
```

Comment:
oh actually this should be <= since if you se MAX_RETRIES = 1 you expect it to be one retry in addition to the original try

Reply:
Yeah, retry_count was more like an num_attempts_count.. Updated here for Batch and in https://github.com/gro-intelligence/api-client/pull/175/files for lib as well